quix-frontend/client/src/lib/runner/services/autocomplete/autocomplete-service.ts

```diff
@@ -46,39 +55,60 @@ export async function setupCompleters(
       query,
       position
     );
+
     const contextCompletions: Promise<AceCompletion[]> = sqlAutocompleter.getCompletionItemsFromQueryContext(
       queryContext
     );
 
     const [keywords, completions] = await Promise.all([
       keywordsCompletions,
-      contextCompletions,
+      contextCompletions
     ]);
 
-    let all =
-      queryContext.contextType === ContextType.Undefined
-        ? keywords
-        : completions;
+    const filteredCompletions: object[] = completions.filter(obj => obj.value.includes(prefix));
+    const searchInObject: Boolean = filteredCompletions.length === 0;
+    if (searchInObject) {
+      nestedColumnCompletions = sqlAutocompleter.getCompletionItemsFromQueryContextColumn(
+        queryContext
+      );
+      [nestedColumnCompletions] = await Promise.all([nestedColumnCompletions]);
+    }
 
-    if (prefix) {
-      const lowerCasedPrefix = prefix.trim().toLowerCase();
+    if (searchInObject && nestedColumnCompletions) {
+      all = queryContext.contextType === ContextType.Undefined ? keywords : nestedColumnCompletions;
+    } else {
+      all = completions;
+    }
 
-      all = all.reduce((resultArr: AceCompletion[], completionItem) => {
-        const indexes = findAllIndexOf(completionItem.value, lowerCasedPrefix);
 
-        if (indexes.length > 0) {
-          completionItem.matchMask = createMatchMask(
-            indexes,
-            lowerCasedPrefix.length
-          );
-          completionItem.score = 10000 - indexes[0];
-          resultArr.push(completionItem);
+    if (prefix) {
+      const lowerCasedPrefix = prefix.trim().toLowerCase();
+      if (searchInObject) {
+        if (prefix.endsWith('.')) {
+          all = enrichCompletionItemAfterDotObject(all);
         }
-
-        return resultArr;
-      }, []);
+        else {
+          all = queryContext.contextType === ContextType.Undefined
+            ? enrichCompletionKeyWord(all, lowerCasedPrefix)
+            : enrichCompletionItemInObjectSearch(all, queryContext, lowerCasedPrefix);
+        }
+      }
+      else {
+        const filterCompletions = all.filter(obj => obj.value.includes(prefix));
+        all = enrichCompletionItemCollumSearch(filterCompletions, lowerCasedPrefix)
+      }
     }
 
+    all.forEach(obj => {
+      const maxLengthOfString = 60;
+      const maxSubStringLength = 57;
```

Comment:
they are different:
in case the length is larger then 60 i would like to take first 57 chars and add three dots.

changed names for better understanding...

Reply:
let's do it differently 

create a util function called `createCaption` next to `createMatchMask`
that gets a string and return the sub string with dots

this function will hold the constants `maxLengthOfString`, `maxSubStringLength`

if the string legth is bigger than the trashold add the dots otherwise return the same string

Reply:
```
(str) => {
    const maxLengthOfString = 60;
    const maxSubStringLength = 57;
      
    if (str.length > maxLengthOfString) {
        return str.substring(0, maxSubStringLength) + "...";
    }

    return str;
}
```

Reply:
got it,
changed the names to 
maxCaptionLength and maxSubCaptionLength